wp-includes/class-wp-service-workers.php

```diff
@@ -57,10 +57,13 @@ class WP_Service_Workers implements WP_Service_Worker_Registry_Aware {
 	 */
 	public function __construct() {
 		$components = array(
-			'configuration'      => new WP_Service_Worker_Configuration_Component(),
-			'navigation_routing' => new WP_Service_Worker_Navigation_Routing_Component(),
-			'precaching_routes'  => new WP_Service_Worker_Precaching_Routes_Component(),
-			'caching_routes'     => new WP_Service_Worker_Caching_Routes_Component(),
+			'configuration'            => new WP_Service_Worker_Configuration_Component(),
+			'navigation_routing'       => new WP_Service_Worker_Navigation_Routing_Component(),
+			'theme_asset_caching'      => new WP_Service_Worker_Theme_Asset_Caching_Component(),
+			'plugin_asset_caching'     => new WP_Service_Worker_Plugin_Asset_Caching_Component(),
+			'uploaded_images__caching' => new WP_Service_Worker_Uploaded_Image_Caching_Component(),
```

Comment:
I'm not sure these should be “components”. They could rather be just global functions that are added to `wp_front_service_worker` in `default-filters.php`. This will allow them to easily be removed or replaced.